### PR TITLE
[#71, #72, #101] Make execution method work with GString

### LINKED ISSUE #71: GString request and 'execute' method fails
Having such a DSL:

```
io.codearte.accurest.dsl.GroovyDsl.make {
                request {
                    method """PUT"""
                    url """/fraudcheck"""
                    body("""
                        {
                        "clientPesel":"${value(client('1234567890'), server(execute('assertThatRejectionReasonIsNull($it)')))}",
                        "loanAmount":123.123
                        }
                    """
                    )
                    headers {
                        header("""Content-Type""", """application/vnd.fraud.v1+json""")

                    }

                }
            response {
                status 200
                body( """{
    "fraudCheckStatus": "OK",
    "rejectionReason": null
}""")
                headers {
                     header('Content-Type': 'application/vnd.fraud.v1+json')

                    }

            }

}
```

results in a following test:

```
def shouldMarkClientAsNotFraud() {
        given:
            def request = given()
                    .header('Content-Type', 'application/vnd.fraud.v1+json')
                    .body('{"clientPesel":"io.codearte.accurest.dsl.internal.ExecutionProperty@362cf66","loanAmount":123.123}')

        when:
            def response = given().spec(request)
                    .put("/fraudcheck")

        then:
            response.statusCode == 200
            response.header('Content-Type') == 'application/vnd.fraud.v1+json'
        and:
            def responseBody = new JsonSlurper().parseText(response.body.asString())
            responseBody.fraudCheckStatus == "OK"
            responseBody.rejectionReason == null
    }
```

### LINKED ISSUE #72: ExecutionProperty is not working in GStrings
For DSL

``` groovy
io.codearte.accurest.dsl.GroovyDsl.make {
                request {
                    method """PUT"""
                    url """/fraudcheck"""
                    body("""
                        {
                        "clientPesel":"${value(client(regex('[0-9]{10}')), server('1234567890'))}",
                        "loanAmount":123.123
                        }
                    """
                    )
                    headers {
                        header("""Content-Type""", """application/vnd.fraud.v1+json""")

                    }

                }
            response {
                status 200
                body( """{
    "fraudCheckStatus": "OK",
    "rejectionReason": ${value(client(null), server(execute('assertThatRejectionReasonIsNull($it)')))}
}""")
                headers {
                     header('Content-Type': 'application/vnd.fraud.v1+json')

                    }

            }

}
```

an exception is thrown while trying to generate Wiremock stubs

```
      at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
        ... 57 more
Caused by: groovy.json.JsonException: Unable to determine the current character, it is not a string, number, array, or object

The current character read is 'i' with an int value of 105
Unable to determine the current character, it is not a string, number, array, or object
line number 3
index number 55
    "rejectionReason": io.codearte.accurest.dsl.internal.ExecutionProperty@26d24d7a
.......................^
        at groovy.json.internal.JsonParserCharArray.decodeValueInternal(JsonParserCharArray.java:216)
        at groovy.json.internal.JsonParserCharArray.decodeJsonObject(JsonParserCharArray.java:140)
        at groovy.json.internal.JsonParserCharArray.decodeValueInternal(JsonParserCharArray.java:196)
        at groovy.json.internal.JsonParserCharArray.decodeValue(JsonParserCharArray.java:166)
        at groovy.json.internal.JsonParserCharArray.decodeFromChars(JsonParserCharArray.java:45)
        at groovy.json.internal.JsonParserCharArray.parse(JsonParserCharArray.java:409)
        at groovy.json.internal.BaseJsonParser.parse(BaseJsonParser.java:103)
        at groovy.json.JsonSlurper.parseText(JsonSlurper.java:208)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoCachedMethodSiteNoUnwrap.invoke(PojoMetaMethodSite.java:210)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:53)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at io.codearte.accurest.dsl.internal.Body.extractValue(Body.groovy:63)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
        at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite$StaticMetaMethodSiteNoUnwrapNoCoerce.invoke(StaticMetaMethodSite.java:148)
        at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite.callStatic(StaticMetaMethodSite.java:99)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:169)
        at io.codearte.accurest.dsl.internal.Body.<init>(Body.groovy:38)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
```

The problem is the same as with `Pattern`. I guess more hacks will have to be added to the `io.codearte.accurest.dsl.internal.Body#extractValue(groovy.lang.GString, groovy.lang.Closure)` ;)

### LINKED ISSUE #101: Execute Command failed to work
Having a DSL with `${value(client(...), server(executeCommand(...)))}` failed to parse the ExecutionCommand part and results in `toString()` resolution.

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
closes #71, closes #72 , closes #101

### CODE REVIEW

**Comment by marcingrzejszczak** [comment]
Don't merge yet ;)

**Comment by marcingrzejszczak** [comment]
Ok - waiting for review (I checked that #71 makes no sense) ;)

**Comment by marcingrzejszczak** [comment]
@szpak @mariuszs @jkubrynski ?

File: accurest-core/src/main/groovy/io/codearte/accurest/util/ContentUtils.groovy [hunk 1]
```diff
@@ -166,18 +173,28 @@ class ContentUtils {
...
 			}
 			return value
 		})
 	}
 
+	static Object returnParsedObject(String string) {
+		Matcher matcher = TEMPORARY_PATTERN_HOLDER.matcher(string.trim())
+		if (matcher.matches()) {
+			List val = matcher[0] as List
+			String pattern = val[1]
```
Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/codearte/accurest/util/ContentUtils.groovy:210]: Think we should check if val contains 2 elements. If not some descriptive exception will be useful

File: accurest-core/src/main/groovy/io/codearte/accurest/util/ContentUtils.groovy [hunk 2]
```diff
@@ -166,18 +173,28 @@ class ContentUtils {
...
+			return Pattern.compile(pattern)
+		}
+		Matcher executionMatcher = TEMPORARY_EXECUTION_PATTERN_HOLDER.matcher(string.trim())
+		if (executionMatcher.matches()) {
+			List val = executionMatcher[0] as List
+			String pattern = val[1]
```
Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/codearte/accurest/util/ContentUtils.groovy:216]: As above

**Comment by jkubrynski** [comment]
:+1: excluding small comments

**Pull request merged**